azurerm/resource_arm_sql_server.go

```diff
@@ -202,3 +239,25 @@ func resourceArmSqlServerDelete(d *schema.ResourceData, meta interface{}) error
 
 	return future.WaitForCompletionRef(ctx, client.Client)
 }
+
+func expandAzureRmSqlServerIdentity(d *schema.ResourceData) *sql.ResourceIdentity {
+	identities := d.Get("identity").([]interface{})
+	identity := identities[0].(map[string]interface{})
+	identityType := sql.IdentityType(identity["type"].(string))
+	return &sql.ResourceIdentity{
+		Type: identityType,
+	}
+}
+
+func flattenAzureRmSqlServerIdentity(identity sql.ResourceIdentity) []interface{} {
```

Comment:
since we need to ensure this is always set (even if there's a bad API response) can we make this:

```suggestion
func flattenAzureRmSqlServerIdentity(identity *sql.ResourceIdentity) []interface{} {
	if identity == nil {
		return []interface{}{}
	}
```

Reply:
Done.